src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackageArgs.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using NuGet.Common;
+using NuGet.Configuration;
+using System.Threading;
+
+namespace NuGet.CommandLine.XPlat
+{
+    public class ListPackageArgs
+    {
+        public ILogger Logger { get; }
+        public string Path { get; set; }
+        public List<PackageSource> PackageSources { get; set; }
+        public IList<string> Frameworks { get; set; }
+        public bool IncludeOutdated { get; set; }
+        public bool IncludeDeprecated { get; set; }
+        public bool IncludeTransitive { get; set; }
+        public bool Prerelease { get; set; }
+        public bool HighestPatch { get; set; }
+        public bool HighestMinor { get; set; }
+        
+        public CancellationToken CancellationToken { get; set; }
+        
+        public ListPackageArgs(
+            string path,
+            List<PackageSource> packageSources,
+            IList<string> frameworks,
+            bool includeOutdated,
+            bool includeDeprecated,
+            bool includeTransitive,
+            bool prerelease,
+            bool highestPatch,
+            bool highestMinor,
+            ILogger logger,
+            CancellationToken cancellationToken)
+        {
+            Path = path ?? throw new ArgumentNullException(nameof(path));
+            PackageSources = packageSources;
```

Comment:
Make sure you do all the appropriate null checks. 
Can this be null?
Can frameworks be null? 
Can the logger be null?

Reply:
No. This class is only used from ListPackageCommand class, and nothing could be passed in as a null from there

Reply:
You still need to do the null checking :)
Null checks are about the callee not the caller.

Reply:
Ok will do